### ISSUE TITLE
Prepare a conda-forge package for easier installation and use of urdf2casadi-matlab

### ISSUE THREAD

**Issue by singhbal-baljinder** [robotology/urdf2casadi-matlab#16]
It would be nice to make urdf2casadi-matlab a [`conda`](https://docs.conda.io/en/latest/) package and make it available on [`conda-forge`](https://conda-forge.org/). An initial reference on how to contribute with new packages on `conda-forge` can be found in https://conda-forge.org/docs/maintainer/adding_pkgs.html.

**Comment by traversaro** [comment]
I am not sure if conda-forge would accept a MATLAB package. Furthermore, this package would need to depend on `casadi-matlab-bindings` that is hosted on the robotology channel, so probably it could make sense to have `urdf2casadi-matlab` package as part of the robotology channel. We can probably just add `urdf2casadi-matlab` to the `ROBOTOLOGY_ENABLE_DYNAMICS_FULL_DEPS` robotology-superbuild profile, this should give us for free the urdf2casadi-conda package.

**Comment by singhbal-baljinder** [comment]
Looking at the [developers-faqs](https://github.com/robotology/robotology-superbuild/blob/73df8adc7bdccbabc3cebc19925db7a041dc64a7/doc/developers-faqs.md#how-to-add-a-new-package) the general steps could be the following:
- write a Build****.cmake file, maybe something like:
~~~
# Copyright (C) Fondazione Istituto Italiano di Tecnologia
# CopyPolicy: Released under the terms of the LGPLv2.1 or later, see LGPL.TXT
include(YCMEPHelper)
include(FindOrBuildPackage)

find_or_build_package(idyntree-matlab-bindings QUIET)
find_or_build_package(casadi-matlab-bindings QUIET)


set(urdf2casadi-matlab_DEPENDS "")
list(APPEND urdf2casadi-matlab_DEPENDS idyntree-matlab-bindings)
list(APPEND urdf2casadi-matlab_DEPENDS casadi-matlab-bindings)

ycm_ep_helper(urdf2casadi-matlab TYPE GIT
              STYLE GITHUB
              REPOSITORY robotology/urdf2casadi-matlab.git
              TAG master
              FOLDER src
              DEPENDS ${urdf2casadi-matlab_DEPENDS})
~~~
- add a find_or_build_package(urdf2casadi-matlab) in https://github.com/robotology/robotology-superbuild/blob/5af37707c0db29cf4ada9209fb3ad3baf6224a59/cmake/RobotologySuperbuildLogic.cmake#L64;
- write a CMakeList.txt in urdf2casadi-matlab, maybe something like the one used in https://github.com/dic-iit/matlab-whole-body-simulator/blob/master/CMakeLists.txt.
- update the doc in https://github.com/robotology/robotology-superbuild/blob/fe67d4b7df6dc72f37f31071e434078c3eddd574/doc/cmake-options.md.

@traversaro Let me know if I am missing something.

**Comment by traversaro** [comment]
That is all correct, the only diffrence is that in the `Build` file you need to do:
~~~
if(ROBOTOLOGY_GENERATE_CONDA_RECIPES)
  find_or_build_package(idyntree-matlab-bindings QUIET)
else()
  find_or_build_package(iDynTree QUIET)
endif()
find_or_build_package(casadi-matlab-bindings QUIET)

set(urdf2casadi-matlab_DEPENDS "")
if(ROBOTOLOGY_GENERATE_CONDA_RECIPES)
  list(APPEND urdf2casadi-matlab_DEPENDS idyntree-matlab-bindings)
else()
  list(APPEND urdf2casadi-matlab_DEPENDS iDynTree)
endif()
list(APPEND urdf2casadi-matlab_DEPENDS casadi-matlab-bindings)
~~~

This is because the `idyntree-matlab-bindings` is a special package that is only existing when  `ROBOTOLOGY_GENERATE_CONDA_RECIPES` is ON, see https://github.com/robotology/robotology-superbuild/blob/9a436af6b921278d0202bd56694725cb1da4cfb9/cmake/RobotologySuperbuildLogic.cmake#L40 .

**Comment by traversaro** [comment]
Furthermore, it may be convenient if `urdf2casadi-matlab` installed its files in a MATLAB namespaces like `+urdf2casadi` or similar, to avoid polluting the MATLAB path with a lot of functions in the global namespace. However, this would probably imply that we need to review our code to add the namespace.
